### DISCUSSION TITLE
Smooth Camera, new launcher, better audio, controllers, bugfixes...

### DISCUSSION THREAD

**Discussion by snouz** [Albeoris/Memoria#402] [Releases]
[COMPLETE CHANGELOG HERE](https://github.com/Albeoris/Memoria/wiki/Changelog-v2024.04.28)

## Features
* Smooth movement: camera moves, scrolling backgrounds and parallax unrestricted from their grid 
* Player camera stabilization (configurable)
* New launcher/Mod manager, 9 new options, 5 new translations
* Better audio quality (new audio backend) and bugfixes
* Native support for PS4, PS5 and Switch controllers
* PSX font embedded
* Widescreen support for 16:10 (e.g. Steamdeck)
* Widescreen support for 19+ more backgrounds
* New cheat: No time limit for Excalibur II
* Anti-Aliasing
* Many gameplay fixes
* Many many visual fixes (widescreen, 60fps, parallax, cropped layers...)
* Many mods-specific features, support, options...

<hr /><em>This discussion was created from the release <a href='https://github.com/Albeoris/Memoria/releases/tag/'>Smooth Camera, new launcher, better audio, controllers, bugfixes...</a>.</em>

**Comment by RickyBO89** [comment]
I've downloaded this new release and something is no more working right, sadly :'(
I had the "skip to main menu" option enabled **before** updating. After updating, I'm sent to menu as expected but no audio is played, both with Moguri soundtrack enabled and disabled. Controller **and** keyboard are totally unresponsive. I've tried to disable the skip to main menu but it keeps skipping to it and, of course, zero audio.

**Reply by Estlelis** [reply]
Same here :(

**Reply by snouz** [reply]
Thanks for reporting, does it still not work when you unplug/replug your controller? What OS are you on? Are you playing with x64 ticked on? Could you share the file "Memoria.log" and "Memoria.ini" in the main directory?

For the moment, I can't reproduce the problem at all, any information can help us fix the problem ASAP.

**Reply by snouz** [reply]
Hi again, I did a hotfix for the release (thanks SamsamTS), could you install this one and tell me if it fixes these problems? (and if not, upload "Memoria.log" and "Memoria.ini"?)
Sorry for the inconvenience!

**Reply by paintin2** [reply]
This fixed it for me, thanks!

**Reply by Estlelis** [reply]
Solved here, thank you!

**Reply by RickyBO89** [reply]
problem solved for me too

**Comment by UncleRemis** [comment]
Something weird has been going on after updating. Every time I load a game, all of my abilities are unequipped. I did a little bit of troubleshooting, but it didn't help.
.
Here's what I did to troubleshoot:
I loaded the game, re-equipped an ability, saved again, exited the game, relaunched the game and loaded the save.

My abilities were all still unequipped. Other changes I make are being saved, but my all my abilities are being unequipped, no matter which save I load.

This has only been happening after I've updated.

Is there anything I can do to fix this? It is an absolute pain in the ass to have to re-equip all of my abilities every time I want to load my save. It kind of makes the game unplayable. I've messed around in the .ini file, but that's not fixing it either.

I've included screenshots of my mods/settings.
![Memoria 1](https://github.com/Albeoris/Memoria/assets/133454966/51368c03-5fd4-48c1-ac12-02aeb74de275)
![Memoria 2](https://github.com/Albeoris/Memoria/assets/133454966/f7dff83d-1c05-4bde-b450-6912f11346a6)

**Reply by SamsamTS** [reply]
Thanks for letting us know. We published a hotfix. Please download and apply the patcher again.

**Reply by UncleRemis** [reply]
That fixed it. I love you.

**Comment by 25klo** [comment]
Question: "ExcaliburIINoTimeLimit (default 0) Give Excalibur II even when time is over", I'm assuming by entering "ExcaliburIINoTimeLimit = 1" will enable this? Assuming 0 (default 0) will keep the original time limit?

**Reply by snouz** [reply]
Exactly, 1 enables the cheat. Fun fact, this can be enabled just before entering the screen where you're supposed to receive it, and it will work.

**Comment by Nyle86** [comment]
Hello everyone
I have the problem that my controller no longer works with the update (XBox)
I can use the joysticks but not the x y a b buttons

**Reply by SamsamTS** [reply]
That's odd. I have a controller that registers as an XBox controller (8BitDo) and it works fine. Maybe try disabling Steam Input.

![image](https://github.com/Albeoris/Memoria/assets/2388532/4bd93624-fc33-4e0f-8b3e-a307f8e57a45)

If this doesn't work, mind sharing your memoria.log?

**Reply by Nyle86** [reply]
Thanks
it works now

**Comment by livnitro** [comment]
hello, i downloaded the update today and i just wanted to ask, is there any way to change the language of the launcher? i read about checking the settings.ini file but i haven't been succesful yet. :(

**Reply by snouz** [reply]
By default, the launcher is set to your computer's language if that translations is included. Currently, we have en, fr, br-pr, de, it, jp, es, ru. If you with to select a different language than what your computer is set to, there's indeed the setting in settings.ini (I forgot the specific setting name), but if that's not recognized, it will default to your computer's language, and if not included, to English.

What language are you trying to set?

**Reply by Albeoris** [reply]
@snouz , 
> but if that's not recognized

hm... no, Settings.ini overrides any auto-resolved locals.

![image](https://github.com/Albeoris/Memoria/assets/4787832/631fca46-bed4-4238-8002-2cd1711ceb40)

So 
```ini
[Memoria]
LauncherLanguage = en
```
must be enough to enforce English.

**Reply by livnitro** [reply]
thanks a bunch !!

**Comment by Rikkuandyuna** [comment]
Hello, I have downloaded the latest memory update, in the notes it says that you can change the gameover background, but I can't get it to work in the game.

It's strange because I changed the image of the home title without any problem. Someone knows something about it? I imagine that the file goes inside here, screenshot of how I put the file in the corresponding path

![Captura de pantalla 2024-05-04 124327](https://github.com/Albeoris/Memoria/assets/168904009/6db79ca7-d7c5-46df-bf2c-c3eb7631ffe0)

**Reply by Albeoris** [reply]
@Rikkuandyuna , please make an issue.

**Reply by Rikkuandyuna** [reply]
How do I do it? Do I open a new topic and ask the question there? sorry but I'm new here

**Reply by snouz** [reply]
Hi @Rikkuandyuna
They're simply in the wrong directory, you need to put these here:
\FINAL FANTASY IX\FF9_Data\EmbeddedAsset\ui\sprites

![image](https://github.com/Albeoris/Memoria/assets/9964576/3fba59dc-720b-40d4-8745-781499217875)
![image](https://github.com/Albeoris/Memoria/assets/9964576/a0d3c63a-862d-4aac-83b1-c5b272cf3c19)

**Reply by Rikkuandyuna** [reply]
Indeed I had it in the wrong location, I listened to it and it worked the first time! thank you so much ^^

And another thing (sorry if I'm annoying) I'm putting new skills in the AbilityFeatures file, and I want to make a skill so that a character automatically uses an item on an ally under a certain circumstance (such as, for example, a phoenix tail when the ally dies, or an antidote when it falls into poison)

I've been practicing for days but I haven't achieved anything. Would you be so kind as to tell me the correct code?

**Reply by Tirlititi** [reply]
I'll see if that's possible and if not I'll add ways to do it. I'm working on adding more options in that file and more control over the possible reactions would definitely be great.

**Reply by Rikkuandyuna** [reply]
Great, I'll keep an eye out for any new modifications you make ;)

**Comment by DragonOfTheEastUA** [comment]
Hello, after the latest update, replacing the game logo does not work. 
I tried two ways "StreamingAssets\UI\Sprites\US\logo.png", and "FF9_Data\EmbeddedAsset\ui\sprites\title_logo.png" and they both don't work.
![FINAL FANTASY IX Screenshot 2024 05 06 - 19 28 50 20](https://github.com/Albeoris/Memoria/assets/169078896/6a255106-0432-46c2-8b21-64c4e6dd3aee)

**Reply by snouz** [reply]
Thanks for the report, Ive just opened https://github.com/Albeoris/Memoria/issues/439
I know where it comes from, we'll fix it in the next update

**Reply by Rikkuandyuna** [reply]
@DragonOfTheEastUA That happened to me once and what I did was restore a backup copy of the game I had, I put the title_logo.png inside steamapps\common\FINAL FANTASY IX\StreamingAssets\UI\Sprites\US, and it worked without problems
![Uploading Captura de pantalla 2024-05-06 200558.jpg…]()

**Reply by DragonOfTheEastUA** [reply]
@Rikkuandyuna No, I tried restoring the vanilla version of the game and installing that version again, but it didn't help.

**Comment by Rikkuandyuna** [comment]
A question to see if anyone knows anything, I've been researching for a long time to change the color of the characters' weapons in battle (e.g. red Artema weapon instead of blue) does anyone know how to do it? I know it is by extracting files with HW, I have these files extracted and their color modified, but I don't know how to add them to the game.

**Reply by Tirlititi** [reply]
You can specify a custom texture in ``Weapons.csv``.
Example:
```txt
#! IncludeHitSfx
#! IncludeCustomTexture
Ultima Weapon;15;5;17;GEO_WEP_B1_026;2;100;0;0;0;376;15;CustomTextures/RedUltimaWeapon.png
```
And then place your custom texture in your mod file with sub-path ``FF9_Data/CustomTextures/RedUltimaWeapon.png``.
That way, you can also have both weapons with blue and red textures as two different weapons.

**Reply by Rikkuandyuna** [reply]
Wow, I did what you said but it didn't work, could it be that I did something wrong? Here I leave screenshots of how I have everything

![Captura de pantalla 2024-05-07 181535](https://github.com/Albeoris/Memoria/assets/168904009/d5851f3b-3828-4dab-b777-66992b9d8b1c)
![Captura de pantalla 2024-05-07 181848](https://github.com/Albeoris/Memoria/assets/168904009/552d39ac-458c-4c0d-9c3e-40f2deeb4927)
![Captura de pantalla 2024-05-07 181942](https://github.com/Albeoris/Memoria/assets/168904009/503bf141-e05a-4409-8c46-d5764c40c8f0)
![Captura de pantalla 2024-05-07 182514](https://github.com/Albeoris/Memoria/assets/168904009/40dce184-889f-40f5-a0e4-0f60f67b9297)

**Comment by Tirlititi** [comment]
EDIT: Sorry, I messed up. This message was meant to be a reply to [that one](https://github.com/Albeoris/Memoria/discussions/402#discussioncomment-9344083) above.

You need the lines
```txt
#! IncludeHitSfx
#! IncludeCustomTexture
```
If you can't have each entry with this format, you can setup these meta-data options and then remove them immediatly:
```txt
# The entry for La Torre here...
#! IncludeHitSfx
#! IncludeCustomTexture
Ultima Weapon;15;5;17;GEO_WEP_B1_026;2;100;0;0;0;376;15;CustomTextures/RedUltimaWeapon.png
#! -IncludeHitSfx
#! -IncludeCustomTexture
# The entry for Sable Ancho here...
```
More infos there: [wiki: externalised-databases](https://github.com/Albeoris/Memoria/wiki/Externalised-databases)

**Reply by Rikkuandyuna** [reply]
I already found the error! Now it works! thank you @Tirlititi !!

**Comment by Rikkuandyuna** [comment]
Hello everyone, here I return again with another question regarding weapons: how can I duplicate them to add different textures and parameters?

For example, copying Vivi's staff, adding more damage points and another weapon texture, but without replacing the original.

I've tried doing it using the Weapons.cs file, but then in combat Vivi always appears equipping Cinna's hammer :(

Sorry if I ask too much, I'm very excited and I want to learn to do more things for my future FFIX mod ;)

**Reply by Tirlititi** [reply]
If you don't want to keep struggling with the ``.csv`` files, I suggest you to try [Hades Workshop](https://forums.qhimm.com/index.php?topic=14315.0). You can edit item and weapon characteristics there and add brand new items as well.

In order to create an new weapon in Hades Workshop, go to the panel "Inventory -> Items -> Regular items" and right-click on a weapon you want to take as a basis, then "Add" and select a numerical ID (whatever number you want; it can be used to identify the item in field scripts and things like that). Then change the weapon's properties to your liking; there's even a field to write your custom texture path there.
In order to make that new weapon accessible, the easiest is to add it to a shop or make an enemy drop it. It could also be given as a sidequest reward or as a treasure in a field, but that's a bit more complicated to do (still, you can do it in Hades Workshop's "Environment -> Fields" panel).

Once that is done, save both with "File -> Save Mod" (this will allow you to resume your changes with Hades Workshop the next time) and with "File -> Save Steam Mod" (this will generate the ``.csv`` and stuff like that automatically).

Beware that when you use "File -> Save Steam Mod", it will delete and overwrite a bunch of files in the folder you pick. Since you started editing multiple files manually before using Hades Workshop, it's better to first save a backup of your mod and/or use "Save Steam Mod" on a new folder temporarily instead of your already existing mod folder.

**Reply by Rikkuandyuna** [reply]
Yes, Hades Workshop is the one I use for the vast majority of things that I have already created, the problem is that when I do what you say and right click on an article, the "Add" and "Remove" options appear gray and I can't do anything, they are like blocked, and the same thing happens to me when trying to create a new support skill (although I don't need to know the latter)

Maybe I need to update HW for it to let me?

Regarding adding these new elements later with the field scripts, I know how to do it ^^

![Captura de pantalla 2024-05-08 131553](https://github.com/Albeoris/Memoria/assets/168904009/c1f7c721-407e-4b50-a2c7-31394bacf831)

**Reply by Tirlititi** [reply]
It might be because you need to download a more recent version of Hades Workshop (the latest should be v0.50b, as said in the "About" window). However, I think it's rather a question of opening a vanilla version of the game instead of a Memoria-modded version.

The best setup for using Hades Workshop in combination with Memoria changed a few years ago. I guess that when you're using Hades Workshop, you use "File -> Open" and then select "FF9_Launcher.exe" of a vanilla copy of the game. The correct practice now (since v0.41) is to select the Memoria-modded version of the game; you basically don't have to make any backup copy of the game at all. When doing so, Hades Workshop will (1) detect that Memoria is available and thus unlock features for which it is required and (2) use the vanilla backup DLL that Memoria created when installing itself.
If (2) fails (because the backup isn't there for some reason or because it's not a backup of the vanilla DLL), there's a long error message when opening the game telling you how to proceed. In your case, it's just a matter of going to your vanilla copy of the game, rename ``Assembly-CSharp.dll`` into ``Assembly-CSharp_Vanilla.dll`` and copy it to the related folder inside the Memoria-patched copy of the game (it's the ``x64`` that is used by HW for this purpose I think). Then you can delete the vanilla copy to only keep the Memoria-patched copy.

**Reply by Rikkuandyuna** [reply]
You are a genius! It worked just as it said, now I can copy weapons and modify them to my liking! thank you so much ;)

By chance you wouldn't also know which script to play to modify the HP that restores regen, or the HP that removes your poison status, right? I would be very interested to know.

**Comment by WarpedEdge** [comment]
Sorry to Hijack, @Rikkuandyuna if you get the chance, i'd love to know the
steps you did from 0 to finish to create new weapons, editing the color or
design, etc. Lots of cool things people make on their own time, but
unfortunately no source of documentation for beginners to follow to start
modding and making their own changes. Creating a document with how you did
all these could make others and myself try their hands at it and start
creating some cool things :) Sorry again to hijack

**Reply by Rikkuandyuna** [reply]
I will try to create a document explaining it, it has not been easy to learn how to do it, we will see if I can explain it well haha ​​;)

**Comment by WarpedEdge** [comment]
Awesome thank you!
